kernel/mellanox_ethernet/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+//! This crate defines the layout of memory objects that make up the software interface between the Mellanox hardware and the driver,
+//! as well as functions to access different fields of these objects.
+//!
+//! The Mellanox ethernet card is referred to as both the NIC (Network Interface Card) and the HCA (Host Channel Adapter).
+//!
+//! All information is taken from the Mellanox Adapters Programmer’s Reference Manual (PRM) [Rev 0.54], unless otherwise specified. 
+
+ #![no_std]
+ #![feature(slice_pattern)]
+ #![feature(core_intrinsics)]
+ #![allow(dead_code)] //  to suppress warnings for unused functions/methods
```

Comment:
Generally this is a bad idea because unused code is typically a design/impl error; if truly needed, put it as an attribute on a specific item instead of a crate-wide attribute.